jwst/cube_build/cube_cloud.py

```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
     spaxel class matched to detector pixels with flux and weighting updated for each
     match
 
-
     """
 
 #________________________________________________________________________________
     if self.instrument == 'MIRI':
-        file_no = file_slice_no
-        det2ab_transform = input_model.meta.wcs.get_transform('detector','alpha_beta')
-        detector2v23 = input_model.meta.wcs.get_transform('detector', 'v2v3')
-        v23toworld = input_model.meta.wcs.get_transform("v2v3","world")
-        worldtov23 = input_model.meta.wcs.get_transform("world","v2v3")
-        v2ab_transform = input_model.meta.wcs.get_transform('v2v3',
-                                                            'alpha_beta')
-
-        alpha, beta, wave = det2ab_transform(x, y)
-        v2, v3, lam23 = detector2v23(x, y)
-        ra,dec,lam = v23toworld(v2,v3,lam23)
-
-        valid1 = np.isfinite(v2)
-        valid2 = np.isfinite(v3)
+        coord1, coord2, wave = input_model.meta.wcs(x, y) # for entire detector find  ra,dec,lambda
+        valid1 = ~np.isnan(coord1)
 
+# MIRI PSF weighting
+# weighting based on distance in alpha-beta plane
         if self.weighting == 'miripsf':
-            wave_resol = self.instrument_info.Get_RP_ave_Wave(this_par1,this_par2)
+            # get transforms needed for this weighting
+
+            det2ab_transform = input_model.meta.wcs.get_transform('detector', 'alpha_beta')
+            worldtov23 = input_model.meta.wcs.get_transform("world", "v2v3")
+            v2ab_transform = input_model.meta.wcs.get_transform('v2v3',
+                                                                'alpha_beta')
+            coord1, coord2, wave = det2ab_transform(x, y)
+            valid1 = ~np.isnan(coord1)
+            wave_resol = self.instrument_info.Get_RP_ave_Wave(this_par1, this_par2)
             alpha_resol = self.instrument_info.Get_psf_alpha_parameters()
             beta_resol = self.instrument_info.Get_psf_beta_parameters()
 
             # transform Cube Spaxel centers to alpha,beta of exposure
-            # for MIRI weighting parameters are based on distance in 
+            # for MIRI weighting parameters are based on distance in
```

Comment:
I'm not sure I understand the logic for MIRI. `coord1` is overwritten several times. 
From the code section about MIRI I think this is what's coded:
```
- if weighting == 'miripsf':
      -- then coord1 is computed in alpha_beta
          get out of if statement
  else:
     -- if alpha_beta is requested:
           -- coord1 is computed in alpha_beta
        else:
          -- coord1 is computed in ra_dec       
```
If this is the desired logic, then it can be coded like this pseudocode avoiding recomputation of `coord1`.

Reply:
I want to pull out the miripsf and alpha-beta weighting since they are special cases.
I will do that in a future pr.